examples/hrtim/eev-comp.rs

```diff
@@ -0,0 +1,143 @@
+
+    let mut delay = cp.SYST.delay(&rcc.clocks);
+
+    let gpioa = dp.GPIOA.split(&mut rcc);
+
+    let input = gpioa.pa1.into_analog();
+    let pin_a: PA8<Alternate<AF13>> = gpioa.pa8.into_alternate();
+
+    let dac1ch1 = dp.DAC1.constrain(dac::Dac1IntSig1, &mut rcc);
+    let mut dac = dac1ch1.calibrate_buffer(&mut delay).enable();
+
+    // Use dac to define the fault threshold
+    // 2^12 / 2 = 2^11 for about half of VCC
+    let limit = 1 << 11;
+    dac.set_value(limit);
+
+    let (comp1, ..) = dp.COMP.split(&mut rcc);
+
+    let comp1 = comp1.comparator(
+        &input,
+        &dac,
+        comparator::Config::default().hysteresis(Hysteresis::None),
+        //.output_inverted(),
+        &rcc.clocks,
+    );
+    comp1.listen(SignalEdge::Rising, &exti);
+    let comp1 = comp1.enable().lock();
+
+    let (mut hr_control, _flt_inputs, eev_inputs) =
+        dp.HRTIM_COMMON.hr_control(&mut rcc).wait_for_calibration();
+
+    let eev_input4 = eev_inputs
+        .eev_input4
+        .bind(&comp1)
+        .edge_or_polarity(external_event::EdgeOrPolarity::Polarity(
+            pwm::Polarity::ActiveHigh,
+        ))
+        .finalize(&mut hr_control);
+
+    let mut hr_control = hr_control.constrain();
+
+    // ...with a prescaler of 4 this gives us a HrTimer with a tick rate of 1.2GHz
+    // With max the max period set, this would be 1.2GHz/2^16 ~= 18kHz...
+    let prescaler = Pscl4;
+
+    //        .               .  *            .
+    //        .  33%          .  *            .               .               .
+    //        .-----.         .--*            .               .-----.         .-----
+    //out1    |     |         |  |            .               |     |         |
+    //        |     |         |  *            .               |     |         |
+    //   ------     -----------  ------------------------------     -----------
+    //        .               .  *            .               .               .
+    //        .               .  *            .               .               .
+    //        .               .  *-------------*              .               .
+    //eev     .               .  |            .|              .               .
+    //        .               .  |            .|              .               .
+    //   -------------------------            .--------------------------------------
+    //        .               .  *            .               .               .
+    //        .               .  *            .               .               .
+    let (mut timer, (mut cr1, _cr2, _cr3, _cr4), mut out1) = dp
+        .HRTIM_TIMA
+        .pwm_advanced(pin_a, &mut rcc)
+        .prescaler(prescaler)
+        .eev_cfg(EevCfgs::default().eev4(EevCfg::default()))
+        .period(0xFFFF)
+        .finalize(&mut hr_control);
+
+    out1.enable_rst_event(&cr1); // Set low on compare match with cr1
+    out1.enable_rst_event(eev_input4);
+    out1.enable_set_event(&timer); // Set high at new period
```

Comment:
@no111u3  sort of related to #98...

I have been experimenting a bit with different ways to specify different types of event sources.

Here an event is specified by passing a reference to the thing to listen to. In this case:
* reset output on `compare register 1` match
* reset output on `external event 4` signal
* set output on timer period event

The nice thing about this is that there is no way to specify the wrong event source by accident. Also there is no need to figure out what enum and variant to to use, just pass in the actual object(or reference to it).

However this requires the event source to exist before the listener which might make things more complicated?

Reply:
@usbalbin thank for this great example. Yesterday I've merged some changes but it will update in some next sessions (for every timers also). So I think about correct and wrong source for event, it might be provided by internal mark field (I'm not sure that rust can do it).

> However this requires the event source to exist before the listener which might make things more complicated?

Following the RM on G4 (and other) family:
>Note: The clock of the slave timer or ADC must be enabled prior to receive events from the master timer, and must not be changed on-the-fly while triggers are received from the master timer.

That means consumer must be created first, I've checked it on my g474 — violation rule makes ADC crazy.